api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -157,97 +169,319 @@ func (n *NemoDatastore) GetStandardLabels() map[string]string {
+					DefaultMode: &initMode,
 				},
 			},
 		},
 		{
-			SecretRef: &corev1.SecretEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: n.Spec.DataStoreParams.EnvSecret,
+			Name: "config",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.DataStoreParams.DataStoreConfigSecret,
+					DefaultMode: &initMode,
+				},
+			},
+		},
+		{
+			Name: "inline-config-sources",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.DataStoreParams.DataStoreInlineConfigSecret,
+					DefaultMode: &configMode,
 				},
 			},
 		},
+		{
+			Name: "temp",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
 	}
+
+	if n.ShouldEnablePersistentStorage() {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: n.Spec.DataStoreParams.Persistence.ClaimName,
+				},
+			},
+		})
+	} else {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	}
+	return volumes
 }
 
-func (n *NemoDatastore) GetInitContainers() []corev1.Container {
-	image := n.Spec.DataStoreParams.InitContainerImage
-	if image == "" {
-		image = "busybox"
-	}
-	cmd := n.Spec.DataStoreParams.InitContainerCommand
-	if len(cmd) == 0 {
-		cmd = []string{
-			"sh",
-			"-c",
-			fmt.Sprintf("until nc -z %s %s; do echo \"PostgreSQL is unavailable. Sleeping for 5 seconds\"; sleep 5; done;", n.Spec.DataStoreParams.DatabaseHost, n.Spec.DataStoreParams.DatabasePort),
-		}
+// GetPVC returns the persistence PVC object for the NeMoDatastore container
+func (n *NemoDatastore) GetPVC() (*corev1.PersistentVolumeClaim, error) {
+	if n.Spec.DataStoreParams.Persistence == nil {
+		return nil, nil
 	}
-	return []corev1.Container{
+
+	persistence := n.Spec.DataStoreParams.Persistence
+
+	size, err := resource.ParseQuantity(persistence.Size)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse size for pvc creation %s, err %v", persistence.ClaimName, err)
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      persistence.ClaimName,
+			Namespace: n.Namespace,
+			Labels:    persistence.Labels,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: persistence.StorageClass,
+			AccessModes:      persistence.AccessModes,
+			Resources: corev1.VolumeResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: size,
+				},
+			},
+		},
+	}, nil
+}
+
+func (n *NemoDatastore) ShouldCreatePersistentStorage() bool {
+	return n.Spec.DataStoreParams.Persistence != nil && n.Spec.DataStoreParams.Persistence.Create
+}
+
+func (n *NemoDatastore) ShouldEnablePersistentStorage() bool {
```

Comment:
i think we don't need `Enabled` field at all, if pvc is nil in the spec, we can assume it is not enabled. If provided in the spec, either an pvc name should to be provided or create option have to be enabled. similar checks are done [here](https://github.com/NVIDIA/k8s-nim-operator/blob/main/internal/controller/nimcache_controller.go#L491).